### ISSUE TITLE
Umbrella: scheduler_perf tests refactor

### ISSUE THREAD

**Issue by kerthcet** [kubernetes/kubernetes#113586]
### What would you like to be added?

We have benchmark tests for scheduler under `test/integration/scheduler_perf`, but I found several problems below:

- [ ] Some util functions are duplicated in e2e tests and integration tests, like `checkWaitListSchedulableNodes`, we should avoid this by moving them to a shared folder.
- [ ] All operation related codes are piled up in [scheduler_perf_test.go](https://github.com/kubernetes/kubernetes/blob/master/test/integration/scheduler_perf/scheduler_perf_test.go), it's hard to maintain, better to split it into smaller ones, like `createNodesOp`, `createNamespacesOp`, etc..
- [ ] Make preparer a singleton object. Currently, when we have multiple `createNodesOp` in `workloadTemplate`, we'll create corresponding instantiations for each of them, it's not necessary, one instantiation is enough(similar to other types of preparer). 
     And we can also benefit from this in cleanup. Nowadays, we'll call node cleanup multiple times(the number equals to `createNodesOp` amount in yaml file), and each time it will delete all nodes, it's ugly. If we implement this, we shall  call cleanup for only once.
- [ ] Make opCode a new definition type https://github.com/kubernetes/kubernetes/pull/113503
- [ ] Add more tests to improve the test coverage in scheduler_perf.
- [ ] Add testcases to cover the performance of pod-unschedulable scenarios.

More(TBD):

- [ ] Implement the delete node operation

All I can think of right now, if we have more TODOs, we can append to the list.

/sig scheduling
/kind feature
/kind cleanup

/assign

### Why is this needed?

For readability and maintainability.

**Comment by ycjiang50** [comment]
Hi @kerthcet, do you need help with any of the tasks?

**Comment by kerthcet** [comment]
Thanks @ycjiang50 , I'd like to discuss about this in the community meeting, but now we're focusing on the v1.26 code freeze stuff, so let's wait for a couple of days if we come to an agreement, I'll let you know.

**Comment by sanposhiho** [comment]
Just for your information: https://github.com/kubernetes/kubernetes/issues/109881

We had some test cases for preemption, but they kept failing and they are currently comment out-ed.
https://github.com/kubernetes/kubernetes/blob/master/test/integration/scheduler_perf/config/performance-config.yaml#L400-L435
https://github.com/kubernetes/kubernetes/pull/109893 We tried to fix it by changing QPS, but actually cannot. 
It'd be nice if we could find a solution on it..

**Comment by tangwz** [comment]
If need any help, I think I can participate.

**Comment by kerthcet** [comment]
/remove-lifecycle rotten

Sig testing is doing some refactor works on this, so I'll defer this work.

**Comment by kerthcet** [comment]
Close this as scheduler_perf changes a lot, we can reticket if necessary.
/close
